### ISSUE TITLE
finetune samples numbers

### ISSUE THREAD

**Issue by yuluoxiadenageyewan** [schwallergroup/matinvent#3]
Is the reason why such a small sample size can initiate fine-tuning due to the introduction of KL regularization and the fact that the number of fine-tuning epochs is relatively small？I previously fine-tuned MatterGen using crystal data with nearly 1,000 migration barriers, but the loss decreased very slowly.

**Comment by jwchen25** [comment]
(1) KL regularization is employed to prevent excessive parameter drift in RL fine-tuning of diffusion model, thereby ensuring fine-tuning stability and preserving the chemical knowledge learned during pre-training. Choosing a smaller sample/evaluation size in RL iterations aims to reduce reward computation costs while maintaining good fine-tuning performance. This is particularly critical for tasks involving expensive reward calculations, such as DFT simulations. Furthermore, our tests demonstrate that 3 epochs per RL iteration are sufficient; too many epochs will increase the time cost of fine-tuning.

(2) If I understand your question correctly, you are fine-tuning MatterGen with adapter using labeled data for conditional generation. First, the loss functions used in RL fine-tuning and MatterGen's conditional fine-tuning are different. Our RL fine-tuning loss is derived from policy gradients computed over the complete 1000-timestep denoising trajectory, whereas the conditional fine-tuning loss typically originates from a single randomly sampled timestep during denoising. This accounts for the higher computational cost of our RL loss calculation.

Furthermore, as demonstrated in our preprint, our RL approach can significantly reduce labeled data requirements and property calculation costs for fine-tuning. One key reason is that RL fine-tuning targets a specific value or the maximization/minimization of a target property. Whereas MatterGen's conditional fine-tuning need to learn the complete conditional probability distribution, the complex mapping between crystal structures and property values across a broad range. Thus, conditional fine-tuning is more difficult and requires more labeled data. However, for specific material applications such as magnet design, we do not need to focus on weakly magnetic materials. Therefore, our RL method can reduce the amount of labeled data required for fine-tuning on specific material tasks.
